includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -349,28 +349,7 @@ protected function read_product_data( &$product ) {
 		$set_props['shipping_class_id'] = current( $this->get_term_ids( $product, 'product_shipping_class' ) );
 		$set_props['gallery_image_ids'] = array_filter( explode( ',', $set_props['gallery_image_ids'] ) );
 
-		if ( '' === $set_props['review_count'] ) {
-			unset( $set_props['review_count'] );
-			WC_Comments::get_review_count_for_product( $product );
-		}
-
-		if ( '' === $set_props['rating_counts'] ) {
-			unset( $set_props['rating_counts'] );
-			WC_Comments::get_rating_counts_for_product( $product );
-		}
-
-		if ( '' === $set_props['average_rating'] ) {
-			unset( $set_props['average_rating'] );
-			WC_Comments::get_average_rating_for_product( $product );
-		}
-
 		$product->set_props( $set_props );
-
-		// Handle sale dates on the fly in case of missed cron schedule.
```

Comment:
Is the handling for broken cron no longer desirable?

Reply:
It *could* indeed be undesiable. However, that is the result of broken cron. Doing this on read each page load is also negative.

One is fixable (cron).
The other negative is not (update on read).

Reply:
Understand. And I think you're right, the correct principle is usually to not work around other people code/incorrect setup in your own codebase (or update objects on read).

However, given that we actually decided to include this in WC back then (and I guess there was a reason for that), I was just wondering what is the reason to remove it now after a year and a half, as a part of unrelated change/PR.

Reply:
The extra write would impact the viability of the lookup table. The lookup table watches the actions to see if 'price' changed. This logic bypases those hooks. 

> However, given that we actually decided to include this in WC back then

Possibly a quick fix introducing tech debt?

Today, if the cron did cause an issue to reoccur we have more options, e.g. we could use actionscheduler for sales instead of cron.

Reply:
Thanks, that makes sense. Let's remove it.